transport-native-epoll/src/main/java/io/netty/channel/epoll/Epoll.java

```diff
@@ -26,36 +26,40 @@
     private static final Throwable UNAVAILABILITY_CAUSE;
 
     static  {
-        Throwable cause = null;
-        FileDescriptor epollFd = null;
-        FileDescriptor eventFd = null;
-        try {
-            epollFd = Native.newEpollCreate();
-            eventFd = Native.newEventFd();
-        } catch (Throwable t) {
-            cause = t;
-        } finally {
-            if (epollFd != null) {
-                try {
-                    epollFd.close();
-                } catch (Exception ignore) {
-                    // ignore
+        if (PlatformDependent.isLinux() && PlatformDependent.bitMode() == 64) {
```

Comment:
This will break custom packaged epoll artifacts.

Reply:
The code supports what the project delivers in it's compiled binaries.

If the project where to deliver 32 but and 64 bit binaries then this would be changed but then also foldering the builds also would be needed eg like how other projects handle native libs, example jnr with jffi

Reply:
I disagree, the same could be said about Netty openssl but the documentation gives instructions on how to build 32 bit artifacts (we actually build an uber jar which works on both 32bit and 64bit architectures). 

http://netty.io/wiki/forked-tomcat-native.html

Reply:
So how is that working to detect the right bit architecture and load the correct compiled lib within the uber? Is it like as I mentioned the jnr approach for jffi, or are you just compiling to 32bit architecture

Reply:
The openssl uber jar uses the architecture as part of the filename in the jar.

Reply:
Cool so it's more akin to what I'm suggesting on the issue comment trace.

Are you able to comment on the issue discussions to answer my one query there?